src/Util/ObjectUtil/ObjectUtil.js

```diff
@@ -11,6 +11,43 @@ import Logger from '../Logger';
  */
 class ObjectUtil {
 
+  /**
+   * Returns the dot delimited path of a given object by the given
+   * key-value pair. Example:
+   *
+   * ```
+   * const obj = {
+   *   level: 'first',
+   *   nested: {
+   *     level: 'second'
+   *   }
+   * };
+   * const key = 'level';
+   * const value = 'second';
+   *
+   * ObjectUtil.getPathByKeyValue(obj, key, value); // 'nested.level'
+   * ```
+   *
+   * Note: It will return the first available match!
+   *
+   * @param {Object} obj The object to obtain the path from.
+   * @param {String} key The key to look for.
+   * @param {String|Number|Boolean} value The value to look for.
+   * @param {String} [currentPath=''] The currentPath (if called in a recursion)
+   *                                  or the custom root path (default is to '').
+   */
+  static getPathByKeyValue(obj, key, value, currentPath = '') {
+    currentPath = currentPath ? `${currentPath}.` : currentPath;
+
+    for (let k in obj) {
```

Comment:
Will this only iterate over enumerable props? IDK, but would prefer this function to only use these keys. Maybe Object keys is an alternative. But as I said, I don't know for sure.

Reply:
Yes, it will only iterater over the enumerbale props. According to [MDN Object.keys()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys) would result in the same behaviour since
>Object.keys() returns an array whose elements are strings corresponding to the enumerable properties found directly upon object.